### ISSUE TITLE
onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion)

### ISSUE THREAD

**Issue by AreaNovi** [jgilfelt/android-sqlite-asset-helper#103]
There's something I didn't understand.
In SQLiteAssetHelper I have the method

```
public void createDBfromAssets(){
        SQLiteDatabase db = this.getWritableDatabase();
        db.close();
    }
```

Then, in my main activity, I check if the DB already exists, otherwise I call the method above in order to initialize the DB

```
//check if a DB already exists, otherwise call a DBHandler method to copy the DB from the assets folder to the internal "databases" folder
File currentDB = new File(this.getDatabasePath(Cons.DB_NAME).getPath());
        if (!currentDB.exists())
            db.createDBfromAssets();
```

I can correctly implement the DB upgrade (I add a new column to one of the DB tables) case by just doing the following things:
1. putting in the asset folder a new DB  for the new users
2. putting in the asset folder the sql file script to upgrade the existing user DB

The problem is that after done this, I need to put values in the new column, depending on the value of other columns. In short, I need to write some code.
**Where do I have to put this code?**

I tried to add to the SQLiteAssetHelper the onUpgrade() method:
```
//code to execute when there is a DB upgrade
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        //from v1 to v2 we added the column "day_free" in the table "days"
        if (oldVersion < 2) {
            updateDaysTable(db);
        }
    }
```

Assuming that the method will be called only in case of a DB upgrade.
Unfortunately, I got this error:
_java.lang.IllegalStateException: attempt to re-open an already-closed object: SQLiteDatabase: <db path>_
the logcat just show me a row where I create an instance of the Database
`SQLiteDatabase db = this.getReadableDatabase();`
If I remove the _onUpgrade()_ method, I don't get the error, but I don't know how to update the values in the upgraded DB.
Any suggestions?
Thanks

**Comment by jon-adams** [comment]
I think you misunderstand the point of this library. The sqlite-asset-helper library takes care of a lot of these steps for you. All you need to do is create the upgrade asset SQL file per the [documentation](https://github.com/jgilfelt/android-sqlite-asset-helper#database-upgrades), name it according to which versions you going from and to, then change the version number in your code. The library takes care of the rest of the boilerplate code and onUpgrade calls for you, running that SQL asset file you put in the correct place per the referenced documentation.

**Comment by Kisty** [comment]
Yeah, I concur with @jon-adams. I ripped out all my code to manually check if database exists already and replaced it with this library by making my database helper extend [`SQLiteAssetHelper`](https://github.com/jgilfelt/android-sqlite-asset-helper/blob/master/library/src/main/java/com/readystatesoftware/sqliteasset/SQLiteAssetHelper.java). Then the library took care of database asset copying and handling.

Have a look at the samples for more of an idea:
https://github.com/jgilfelt/android-sqlite-asset-helper/tree/master/samples/database-v1
https://github.com/jgilfelt/android-sqlite-asset-helper/tree/master/samples/database-v2-upgrade
